UI Development/gui.pde

```diff
@@ -172,6 +181,23 @@ class Gui {
         tempPage.draw();
       }
     }
+    
+    if (searchFilters.size() != 0 )
+    {
+      if (searchFilters.contains("airports"))
+      {
+        airport.draw();
+      }
+      if (searchFilters.contains("dates"))
+      {
+        date.draw();
+      }
+      if(searchFilters.contains("delays"))
+      {
+        delay.draw();
+      }
+    }
```

Comment:
This if-block should practically be put in the ChartPage class, as it deals with the back-end.

For example, notice how no button or searchbar object is drawn in the Gui top-level class. Instead, both types of objects are drawn in the `draw()` of the Page they're on. This has been done for the purpose of encapsulation. Gui must **only** take care of drawing the pages.

Applying the same principle, any back-end class used for the purpose of charting should have its `draw()` called within the `draw()` of the ChartPage class.